ocamltest/Makefile

```diff
@@ -176,6 +176,7 @@ directories := $(addprefix $(ROOTDIR)/,utils bytecomp parsing stdlib \
 include_directories := $(addprefix -I , $(directories))
 
 flags := -g -nostdlib $(include_directories) \
+  -ccopt -I$(ROOTDIR)/runtime \
```

Comment:
I'm curious about why this is necessary and what exactly it does. My assumption is that the semantics of `-output-complete-obj` is to link with the runtime to produce a C executable, so I guess at compile-time the C compiler needs to go find the runtime, and without this `-ccopt` it fails on systems where OCaml is not already installed.

But this leaves the following questions (in the context that it's important to ensure we never implicitly depend on files outside the source repository, during the test of the compiler, to avoid being polluted by existing installations):

- do we know for sure that this `-I` option will take priority over wherever else the compiler would naturally go for a runtime to link?
- is there a way to tell `-output-complete-obj` to *not* look at the standard installation place (which may be stale from an experimental-compiler-build perspective)? Does `-nostdlib` suffice to get this semantics?

Reply:
For *compiling*, `-nostdlib` means *not* passing `-I <stdlib>` to the *C* compiler. So the above `-ccopt` flag "fixes" this (the runtime headers are necessary to compile the C file generated by `-output-complete-exe`).

For *linking*, it is a different story as every `-I` flag passed to the *OCaml* compiler gets passed as a `-L` flag to the linker (see `Ccomp.call_linker`).

(In passing, I wonder why we don't pass `-I` flags passed to the OCaml compiler on to the C compiler...)

Reply:
Your first point answered my main question: `-nostdlib` does correctly ensure that the installation-site runtime will not be picked by `-output-complete-exe`.